### DISCUSSION TITLE
Magepack Generate Does Not Work

### DISCUSSION THREAD

**Discussion by sprankhub** [wardenenv/warden#661] [Help & Support]
If I try to run the `magepack generate` command, I do not get any output and it hangs forever:

    $ warden env exec magepack magepack generate --cms-url="https://app.shop.test/cms-url" --category-url="https://app.shop.test/category-url" --product-url="https://app.shop.test/product-url"
    ℹ Collecting bundle modules in the browser.
    ℹ Collecting modules for bundle "category".

When I log into the container via `docker exec -it shop_magepack_1 sh` and run Magepack with the debug flag, I get this:

    $ magepack generate --cms-url="https://app.shop.test/cms-url" --category-url="https://app.shop.test/category-url" --product-url="https://app.shop.test/product-url" --debug

     ERROR  Failed to launch the browser process!                                                                                                                                                            12:45:11
    [176:192:0619/124511.512146:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
    [176:176:0619/124511.515389:ERROR:ozone_platform_x11.cc(239)] Missing X server or $DISPLAY
    [176:176:0619/124511.515399:ERROR:env.cc(255)] The platform failed to initialize.  Exiting.

    TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/master/docs/troubleshooting.md

      [176:192:0619/124511.512146:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
      [176:176:0619/124511.515389:ERROR:ozone_platform_x11.cc(239)] Missing X server or $DISPLAY
      [176:176:0619/124511.515399:ERROR:env.cc(255)] The platform failed to initialize.  Exiting.
      
      
      TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/master/docs/troubleshooting.md
      
      at onClose (/usr/local/lib/node_modules/magepack/node_modules/puppeteer/lib/Launcher.js:750:14)
      at ChildProcess.<anonymous> (/usr/local/lib/node_modules/magepack/node_modules/puppeteer/lib/Launcher.js:740:61)
      at ChildProcess.emit (node:events:525:35)
      at ChildProcess._handle.onexit (node:internal/child_process:291:12)

Can someone reproduce that? Does someone have any idea how this could be fixed?

**Comment by navarr** [comment]
When attempting to run the first command, it executed successfully in my test environment with no issues.

What version of MagePack are you running?  In my case, it was `2.11.2`. (`warden env exec magepack magepack -v`)

**Reply by sprankhub** [reply]
It is also `2.11.2` for me. I also ran `warden env pull && warden env up`. Any idea? It is quite strange as the error message does not really sound environment-specific to me :thinking:

**Reply by bap14** [reply]
@sprankhub  Well `warden env pull` will pull the latest built images for your services, so it's possible whatever version you had cached locally had something wrong with it, and it was fixed in a later image release, but that's just a guess.

**Comment by sprankhub** [comment]
So.... It looks like it works with Warden 0.14.0! It could be that I used a more out-of-date version of Warden due to an old manual installation. Thanks for your help, @navarr!

**Comment by sprankhub** [comment]
Even though it worked directly after the upgrade, it stopped working again now even with Warden `0.14.0` and updated images. The error is still the same. Any ideas?

**Reply by navarr** [reply]
Did this ever start working again?  It's certainly a novel error

**Reply by sprankhub** [reply]
Nope, I still have exactly that issue.

**Reply by ihor-sviziev** [reply]
Since it's using nodejs, maybe you've changed version of nodejs? On which one you're trying to run?

**Reply by sprankhub** [reply]
Thanks for the feedback, @ihor-sviziev and sorry for the very late answer. 

    $ node --version
    v18.17.0

Magepack just requires anything greater 10. @navarr, which Node version did you use during testing?

**Reply by ihor-sviziev** [reply]
@sprankhub I see on Magento 2.4.6 there were changes in packages.json.sample file. I'm currently running 2.4.5-p-something, maybe that's the reason it works fine for me?
Could say what the Magento version you're using and could you try using this file?
https://github.com/magento/magento2/blob/2.4.5-p4/package.json.sample

**Reply by sprankhub** [reply]
I could finally continue working on this. Unfortunately, it still does not work. I do not think it has anything to do with the `npm` dependencies. The Magepack container looks weird to me. For instance:

    $ ls -alh /var/run/
    total 8K     
    drwxr-xr-x    2 root     root        4.0K Sep 28 11:18 .
    drwxr-xr-x    1 root     root        4.0K Oct 16 13:45 ..

Is this the same for you?

I also cannot find any logs inside this container:

    $ ls -alh /var/log/
    total 8K     
    drwxr-xr-x    2 root     root        4.0K Sep 28 11:18 .
    drwxr-xr-x    1 root     root        4.0K Oct 16 06:21 ..

**Comment by janssensjelle** [comment]
Experiencing the same issue here. Hangs after 


```
ℹ Collecting modules for bundle "category".                                                                                 07:24:40
```

did anyone solve this?

**Reply by sprankhub** [reply]
Which version of Magepack do you use? Did you try the debugging steps outlined above?
